src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -984,36 +922,45 @@ public float CalculateSpeed()
 
     public float CalculateHitpoints()
     {
-        var maxHitpoints = Membrane.Hitpoints +
-            (Rigidity * Constants.MEMBRANE_RIGIDITY_HITPOINTS_MODIFIER);
+        var maxHitpoints = CurrentSpecies.MembraneType.Hitpoints +
+            (CurrentSpecies.MembraneRigidity * Constants.MEMBRANE_RIGIDITY_HITPOINTS_MODIFIER);
 
         return maxHitpoints;
     }
 
     /// <summary>
     ///   Returns the cost of the organelle that is about to be placed
     /// </summary>
-    public float CalculateCurrentOrganelleCost()
+    public int MutationPointsAfterOrganelleAddition()
     {
+        if (MovingOrganelle != null)
+        {
+            return Constants.BASE_MUTATION_POINTS - Constants.ORGANELLE_MOVE_COST;
```

Comment:
This isn't entirely dynamic right? It should be detected that the organelle is being moved to the original position as that should cost 0 right as the organelle will match the starting species.